### ISSUE TITLE
ARG flake8-unused-args does not work with dunder methods

### ISSUE THREAD

**Issue by alexruddick-grantek** [astral-sh/ruff#21904]
### Summary

My initial concern was finding (bad) code that tried to pass `*args` to `__aenter()__`.  
The `*args` has no effect, and should be caught by a linter IMO. 
https://docs.python.org/3/reference/datamodel.html#object.__aenter

The maintainers at `flake8-asyncio` [pointed out](https://github.com/python-trio/flake8-async/issues/421#event-21464789729) that this is not only just `__aenter__` but also affects a number of other dunder methods.

I also realized there is flake8 plugin that catches this: `flake8-unused-arguments`.  `ruff` implements these rules as `ARG`.  After doing some testing, I realized that `ruff` doesn't flag unused arguments in dunder methods _at all_.  Not `some_arg`, not `*args`, and not `**kwargs`.

```python
class Class:
    def foo(self, arg1, arg2):
        print(arg1)  # ARG002 Unused method argument: `arg2`

    def __aenter__(self, unused):  # ruff does not error, flake8 does
        return self

class Foo:
    async def __aenter__(self, *args):   # ruff does not error, flake8 does
        return self

class Bar:
    async def __aenter__(self, **kwargs):   # ruff does not error, flake8 does
        return self

class Baz:
    async def __aenter__(self):  # OK
        return self
```
See reproduction repo here: https://github.com/alexrudd2/unused-args (ruff + flake8) or [playground](https://play.ruff.rs/7fa84f6b-b5cf-43fb-8fde-2a3c3dfd48c0).

### Version

0.14.8

**Comment by alexrudd2** [comment]
(posting here to subscribe to notifications on my intended account)

**Comment by amyreese** [comment]
I assume the ARG rule in ruff is just trying to avoid needing a catalog of 100+ dunder methods to know which ones have arguments that need to be used or not, as it is common to not use arguments in methods like `__exit__` or `__aexit__` for decorators that don't care about exception state, etc.

**Comment by ntBre** [comment]
Related: https://github.com/astral-sh/ruff/issues/1796, the corresponding PR is where the exception for dunder methods was added.
